src/utils/Validator.ts

```diff
@@ -116,6 +125,23 @@ const validateMetric = (key: string, value: number, unit?: Unit | string): void
   ) {
     throw new InvalidMetricError(`Metric unit ${unit} is not valid`);
   }
+
+  if (
+    storageResolution !== undefined &&
+    !Object.values(StorageResolution)
+      .map((s) => s)
+      .includes(storageResolution)
+  ) {
+    throw new InvalidMetricError(`Metric resolution ${storageResolution} is not valid`);
+  }
+
+  if (metricNameAndResolutionMap && metricNameAndResolutionMap.has(key)) {
```

Comment:
Why are we checking if `metricNameAndResolutionMap` is null/undefined?

Reply:
Yes, not needed. Removed it!